cookbooks/bcpc-hadoop/recipes/disks.rb

```diff
@@ -47,17 +47,28 @@
       # What disks will bcpc and bcpc-hadoop feel free to blank?
       # By default, all sd* devices (excluding sda) and all md* devices.
       #
-      # On our EFI-based VM builds, it's very important to omit sdb, as
-      # that is the 32 MB image containing iPXE.  (It's relatively
-      # harmless to overwrite it, but it will cause graphite to fail when
-      # /disk/0 fills up.)
+      # On our EFI-based VM builds, it's very important to the 32 MB
+      # image containing iPXE.  (It's relatively harmless to overwrite
+      # it, but it will cause graphite to fail when /disk/0 fills up.)
+      #
+      # We also reject any block device we are unable to open with O_EXCL,
+      # because that means it is already in use by the kernel.
       #
       all_drives = node[:block_device].keys.select do |dd|
-        dd =~ /sd[a-i]?[b-z]/ ||
-        dd =~ /md\d+/
+        dd =~ /sd[a-i]?[a-z]/ || dd =~ /md\d+/
```

Comment:
The minor number in a device file is 8 bits, so the last possible device node, (135, 255), will be /dev/sdiv 

https://github.com/torvalds/linux/blob/v4.12/Documentation/admin-guide/devices.txt#L2067-L2072